server/statistics/buckets/bucket_stat_informer.go

```diff
@@ -26,6 +46,172 @@ type BucketStat struct {
 	EndKey    []byte
 	HotDegree int
 	Interval  uint64
-	// see statistics.RegionStatKind
+	// the order should see statistics.RegionStatKind
 	Loads []uint64
 }
+
+func (b *BucketStat) clone() *BucketStat {
+	c := &BucketStat{
+		RegionID:  b.RegionID,
+		HotDegree: b.HotDegree,
+		Interval:  b.Interval,
+		StartKey:  make([]byte, len(b.StartKey)),
+		EndKey:    make([]byte, len(b.EndKey)),
+		Loads:     make([]uint64, len(b.Loads)),
+	}
+	copy(c.Loads, b.Loads)
+	copy(c.StartKey, b.StartKey)
+	copy(c.EndKey, b.EndKey)
+	return c
+}
+
+func (b *BucketStat) String() string {
+	return fmt.Sprintf("[region-id:%d][start-key:%s][end-key-key:%s][hot-degree:%d][Interval:%d(ms)][Loads:%v]",
+		b.RegionID, core.HexRegionKeyStr(b.StartKey), core.HexRegionKeyStr(b.EndKey), b.HotDegree, b.Interval, b.Loads)
+}
+
+// BucketTreeItem is the item of the bucket btree.
+type BucketTreeItem struct {
+	regionID uint64
+	startKey []byte
+	endKey   []byte
+	stats    []*BucketStat
+	interval uint64
+	version  uint64
+	status   status
+}
+
+// GetStartKey returns the start key of the bucket tree.
+func (b *BucketTreeItem) GetStartKey() []byte {
+	return b.startKey
+}
+
+// GetEndKey return the end key of the bucket tree item.
+func (b *BucketTreeItem) GetEndKey() []byte {
+	return b.endKey
+}
+
+// String implements the fmt.Stringer interface.
+func (b *BucketTreeItem) String() string {
+	return fmt.Sprintf("[region-id:%d][start-key:%s][end-key:%s]",
+		b.regionID, core.HexRegionKeyStr(b.startKey), core.HexRegionKeyStr(b.endKey))
+}
+
+// Less returns true if the start key is less than the other.
+func (b *BucketTreeItem) Less(than btree.Item) bool {
+	return bytes.Compare(b.startKey, than.(*BucketTreeItem).startKey) < 0
+}
+
+// equals returns whether the key range is overlaps with the item.
+func (b *BucketTreeItem) equals(origin *BucketTreeItem) bool {
+	if origin == nil {
+		return false
+	}
+	// key range must be same if the version is same.
+	if b.version == origin.version {
+		return true
+	}
+	return bytes.Equal(b.startKey, origin.startKey) && bytes.Equal(b.endKey, origin.endKey)
+}
+
+// cloneBucketItemByRange returns a new item with the same key range.
+// item must have some debris for the given key range
+func cloneBucketItemByRange(b *BucketTreeItem, startKey, endKey []byte) *BucketTreeItem {
```

Comment:
Why not make this become a method of `b *BucketTreeItem`?

Reply:
fix